### PR TITLE
[OYPD-273] updating embed media styles

### PR DESCRIPTION
**Pull request by david-hernandez**
Jira issue: https://propeople-us.atlassian.net/browse/OYPD-273

- [ ] Add a node of some kind. For example, blog post.
- [ ] Add embedded media using the WYSIWYG.
- [ ] Verify the different embed options look and work correctly.

- Images have full size, half size (maximum 470px width), and link. They can be set to none, left, center, or right.
- Video will be a natural full width or a link.
- Document will be a natural size for the document embed. Link can also be used. They can be set to none, left, center, or right.
- All links will open in a new tab.

### CODE REVIEW

**Comment by david-hernandez** [comment]
So I'm not sure what the intention was for documents and videos with regard to the half size display mode. They do not have a defined width like images do. The documents will have whatever width the iframe is and similar for the video. They can't really be floated without an actual size and I we don't have a size to go width because this is a site wide feature.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build530

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/530/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build530/profiles/openy/build/reports/behat

**Review by ddrozdik — APPROVED** [review]

**Comment by ddrozdik** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build531

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/531/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build531/profiles/openy/build/reports/behat

**Comment by alexschedrov** [comment]
## Half size for image on one column layout.
When I select half view mode on one column layout for image, it's not exactly half size.
image

## Half size for document and video.
> So I'm not sure what the intention was for documents and videos with regard to the half size display mode. They do not have a defined width like images do.

They do. We should set them 50% for the width so, they can have left/right alignment.
image

image

## Full view doesn't work for document
image


## Half size for document
That's how embedded video should look like with half view mode
image
image

**Comment by david-hernandez** [comment]
That last commit should fix the half size image. My math was wrong. I had it at 470, instead of 570.

screen shot 2017-02-21 at 10 16 49 am

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build541

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/541/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build541/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build542

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/542/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build542/profiles/openy/build/reports/behat

**Comment by david-hernandez** [comment]
The document is embedded with an iframe, but has no defined height or width defined so the plugin that loads the actual document loads its initial size based on the container. I don't know of any way to handle that other than hard coding the size. The document itself won't be responsive.

screen shot 2017-02-21 at 10 28 19 am
screen shot 2017-02-21 at 10 26 57 am

The half size won't be sorted until we figure out how to get context for the view mode to the wrapper . I tried playing around a bit with hard coding sizes as well, but it doesn't play well with being responsive.

**Comment by david-hernandez** [comment]
Ok I found out that in preprocess we have access to the context for the view mode in order to add that as a class on the wrapper. I did that and updated the styling. The images and video seem to work pretty well now. Document is still sketchy and requires a defined size, but I can at least get left and right to work with wrapper text.

screen shot 2017-02-21 at 12 35 03 pm
screen shot 2017-02-21 at 12 31 38 pm

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>2</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build548/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build548

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/548/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build548/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>2</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build554/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build554

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/554/

File: themes/openy_themes/openy_rose/openy_rose.theme [hunk 1]
```diff
@@ -104,3 +105,19 @@ function openy_rose_preprocess_field__node__field_location_directions(&$variable
...
+ * Prepares variables for entity embed container templates.
+ *
+ * Default template: entity-embed-container.html.twig.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - element: An associative array containing the properties of the element.
+ *     Properties used: #attributes, #children.
+ */
+function openy_rose_preprocess_entity_embed_container(&$variables) {
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/openy_rose.theme]: Is it possible to move this preprocess to the openy_media? So it could be part of package.

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build554/profiles/openy/build/reports/behat

**Comment by alexschedrov** [comment]
It looks amazing!

1. Please fix code sniffer notices
2. Please move preprocess to `openy_media` module if possible.

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build557

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/557/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build557/profiles/openy/build/reports/behat

**Pull request merged**